spring-cloud-vault-spring-connector/src/main/java/io/pivotal/spring/cloud/vault/config/java/VaultServiceInfoPropertySourceAdapter.java

```diff
@@ -18,21 +18,28 @@
 import java.util.HashMap;
 import java.util.Map;
 
-import io.pivotal.spring.cloud.vault.service.common.VaultServiceInfo;
-
 import org.springframework.core.env.MapPropertySource;
 import org.springframework.core.env.PropertySource;
 
+import io.pivotal.spring.cloud.vault.service.common.VaultServiceInfo;
+
 /**
  * Overriding property to configure
  * {@link org.springframework.cloud.vault.config.VaultProperties} via
  * {@link VaultServiceInfo} property exposure.
  *
  * @author Mark Paluch
  */
-class VaultServiceInfoPropertySourceAdapter extends
+class VaultServiceInfoPropertySourceAdapter extends 
 		ServiceInfoPropertySourceAdapter<VaultServiceInfo> {
 
+	private static final Map<String, Integer> schemeDefaultPort = new HashMap<>();
+
+	static {
+		schemeDefaultPort.put("http", 80);
```

Comment:
Is there a link that describes these assumptions? Vault's default port is `8200` for HTTP and HTTPS.

Reply:
I made only one assumption about port 80. Vault by default supports only HTTPS. I do not know it it  supports HTTP at all in any version. Problem with this issue if port is not configured explicitly in vault service broker. 8200 always will be explicitly added to vault address. Defaults for HTTP - 80 and for HTTPS - 443, this is about those cases only, when port is determined based on protocol not from URL.